services/identity/src/main/java/com/crapi/controller/AuthController.java

```diff
@@ -48,11 +49,19 @@ public class AuthController {
      */
     @PostMapping("/login")
     public ResponseEntity<JwtResponse> authenticateUser(@Valid @RequestBody LoginForm loginForm) throws UnsupportedEncodingException {
-        JwtResponse jwtToken = userService.authenticateUserLogin(loginForm);
-        if (jwtToken!=null && jwtToken.getToken()!=null) {
-            return ResponseEntity.status(HttpStatus.OK).body(jwtToken);
-        }else {
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(jwtToken);
+        try {        
+            
+            JwtResponse jwtToken = userService.authenticateUserLogin(loginForm);
+            
+            if(jwtToken.getToken() != null) {
+                return ResponseEntity.status(HttpStatus.OK).body(jwtToken);
+            }
+            else {
+                return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(jwtToken);
+            }
+
+        } catch (BadCredentialsException e) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(new JwtResponse("",UserMessage.INVALID_CREDENTIALS));
```

Comment:
Instead of try catch we should handle the null scenario of JWT token.

Reply:
The if block checking for the null condition of the JWT token is part of the try block.